torch/csrc/jit/fuser/executor.cpp

```diff
@@ -372,10 +363,33 @@ bool runFusion(const int64_t key, Stack& stack) {
   }
   maybe_kernel = spec.findKernel(arg_spec);
   AT_ASSERT(maybe_kernel);
+  return KernelInfo{*maybe_kernel, inputs, device};
+}
+
+bool runFusion(const int64_t key, Stack& stack) {
+  // Short-circuits if fusion isn't enabled
+  if (!canFuseOnCPU() && !canFuseOnGPU())
+    return false;
+
+  // Acquires the FusionSpec
+  auto maybe_spec = retrieve(key);
+  AT_ASSERT(maybe_spec);
+  auto& spec = *(*maybe_spec);
+
+  // Acquires inputs from stack
+  auto all_inputs = last(stack, spec.nInputs());
+
+  std::shared_ptr<FusedKernel> kernel;
+  std::vector<at::Tensor> inputs;
+  at::Device device = at::kCPU;
+  std::tie(kernel, inputs, device) = getKernelFromArgs(all_inputs, spec);
```

Comment:
The boundary between runFusion and this function is ill defined and confusing. Can you refactor this to make it more clear. Before it made sense because it was self contained to the function. Do you really need have this be a separate function? Can't we just do the same debug code by having runFusion take  Graph* that it fills in with the graph it used?

Reply:
I don't understand your last sentence. The reason we're doing this is to get the FusedKernel and either run it or get its C++ source code. That doesn't have anything to do with graphs??